app/models/concerns/error_trackable.rb

```diff
@@ -0,0 +1,41 @@
+module ErrorTrackable
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def setup_error_threshold(threshold, trigger_method, reset_on = nil)
+      class_attribute :error_threshold, :error_trigger_method
+
+      self.error_threshold = threshold
+      self.error_trigger_method = trigger_method
+
+      return if reset_on.blank?
+
+      after_commit :reset_error_count, on: reset_on
+    end
+  end
+
+  def track_error
+    ::Redis::Alfred.incr(error_count_key)
+    send(error_trigger_method) if threshold_breached?
+    Rails.logger.info "[ErrorTrackable] Tracking error ##{error_count} for #{self.class.table_name.singularize} with id: #{id}"
+  end
+
+  def error_count
+    ::Redis::Alfred.get(error_count_key).to_i
+  end
+
+  def threshold_breached?
+    error_count >= error_threshold
+  end
+
+  private
+
+  def error_count_key
+    format(::Redis::Alfred::ERROR_TRACKABLE_COUNT, obj_type: self.class.table_name.singularize, obj_id: id)
```

Comment:
This PR changes the key to a standard one across the app, the Reauthorizable had a specific key. Is this fine? @sojan-official